packages/builders/src/components/selectMenu/UnsafeSelectMenu.ts

```diff
@@ -1,45 +1,60 @@
-import { ComponentType, type APISelectMenuComponent } from 'discord-api-types/v9';
-import type { Component } from '../Component';
-import { SelectMenuOption } from './SelectMenuOption';
+import { APISelectMenuOption, ComponentType, type APISelectMenuComponent } from 'discord-api-types/v9';
+import { Component } from '../Component';
+import { UnsafeSelectMenuOption } from './UnsafeSelectMenuOption';
 
 /**
  * Represents a non-validated select menu component
  */
-export class UnsafeSelectMenuComponent implements Component {
-	public readonly type = ComponentType.SelectMenu as const;
-	public readonly options: SelectMenuOption[];
-	public readonly placeholder?: string;
-	public readonly min_values?: number;
-	public readonly max_values?: number;
-	public readonly custom_id!: string;
-	public readonly disabled?: boolean;
-
-	public constructor(data?: APISelectMenuComponent) {
-		this.options = data?.options.map((option) => new SelectMenuOption(option)) ?? [];
-		this.placeholder = data?.placeholder;
-		this.min_values = data?.min_values;
-		this.max_values = data?.max_values;
-		/* eslint-disable @typescript-eslint/non-nullable-type-assertion-style */
-		this.custom_id = data?.custom_id as string;
-		/* eslint-enable @typescript-eslint/non-nullable-type-assertion-style */
-		this.disabled = data?.disabled;
+export class UnsafeSelectMenuComponent extends Component<Omit<APISelectMenuComponent, 'options'>> {
+	public readonly options: UnsafeSelectMenuOption[] = [];
+
+	public constructor(
+		data?: APISelectMenuComponent & { type?: ComponentType.SelectMenu; options?: APISelectMenuOption[] },
+	) {
+		super({ type: ComponentType.SelectMenu, ...data });
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+		this.options = data?.options?.map((o) => new UnsafeSelectMenuOption(o)) ?? [];
```

Comment:
Same as `ActionRow`, `options` is being passed to super, which makes `usmc.data.options` and `usmc.options` both exist with mismatching values after using `usmc.addOptions()` or `usmc.setOptions()`.

The way to fix this is by changing the `data` parameter to `{ options, ...data }`.